regular_shapes.scad

```diff
@@ -154,9 +157,11 @@ module pentagon_tube(height,radius,wall)
  tubify(radius,wall) pentagon_prism(height,radius);
 }
 
-module hexagon_prism(height,radius)
+module hexagon_prism(height, radius, across_flats)
 {
-  linear_extrude(height=height) hexagon(radius);
+  r = radius;
+  r = across_flats ? across_flats/2/cos(30) : radius;
+  linear_extrude(height=height) hexagon(r);
```

Comment:
Just use `hexagon(radius=radius, across_flats=across_flats)` and let the other module do the heavy work.

Reply:
Done. Looks much better now.